src/android/io/cozy/plugins/listlibraryitems/ListLibraryItems.java

```diff
@@ -421,7 +421,16 @@ protected void onProgressUpdate(Integer... values) {
 
             Log.i("", "Wrote " + values[0] + " bytes (total: " + values[1] + " / " + values[2] + ")");
 
-            // TODO: send progress
+            float percent = ((float) values[1] / (float) values[2]) * 100.f;
+            JSONObject json_progress = new JSONObject();
+            try {
+                json_progress.put("progress",(int)percent);
+            } catch (JSONException ex) {
+
+            }
+            PluginResult pr = new PluginResult(PluginResult.Status.OK, json_progress);
+            pr.setKeepCallback(true);
+            mCallback.sendPluginResult(pr);
```

Comment:
I don't think this is going to work too well, because I believe `mCallback.sendPluginResult` can only be called once — when the upload is over. It triggers the [`onSuccess` call](https://github.com/cozy/cordova-plugin-list-library-items/blob/master/www/ListLibraryItems.js#L23) here, and then the next calls to `mCallback.sendPluginResult` will never go through to javascript.

Can you check if I'm wrong on this please? Essentially replacing the line I pointed out by `cordova.exec(console.log, onError, 'ListLibraryItems', 'uploadItem', [payload]); should tell you what the JS sees.
 `

Reply:
I was wrong! It works.